biosignals.py

```diff
@@ -0,0 +1,74 @@
+import platform
+import sys
+from dataclasses import dataclass, field
+
+import plux
+
+osDic = {"Darwin": "MacOS",
+         "Linux": "Linux64",
+         "Windows": ("Win32_37", "Win64_37")}
+if platform.system() != "Windows":
+    sys.path.append("PLUX-API-Python3/{}/plux.so".format(osDic[platform.system()]))
+else:
+    if platform.architecture()[0] == '64bit':
+        sys.path.append("PLUX-API-Python3/Win64_37")
+    else:
+        sys.path.append("PLUX-API-Python3/Win32_37")
+
+
+@dataclass
+class Device:
+    """
+    Stores the integer corresponding to the port occupied by the device.
+    Defaults to port 1
+    """
+
+    source: str
+
+    def set_port(self, source: int):
+        self.source = source
+
+
+@dataclass
+class Session(Device):
+    """
+    Stores session data in a dict
+    """
+    Device: Device
+    Session_Data: dict
+    sampling_frequency: int = 1200
+    Resolution: int = 16
+    Allowed_Resolutions: list = field(default_factory=lambda: [8, 16])
+    Allowed_Settings: list = field(default_factory=lambda: ['fs', 'res', 'source'])
+
+    def prepare_dict(self):
+        """
+        Initializes the dictionary where the session's settings
+        are stored.
+        """
+        self.Session_Data = {
+            'fs': self.sampling_frequency,
+            'source': Device.source,
+            'res': self.Resolution
+        }
+
+    def update_dict(self, setting: str, value: int):
+        if setting not in self.Allowed_Settings:
+            raise ValueError('The setting introduced is not allowed')
+        else:
+            if setting == 'fs':
+                self.sampling_frequency = value
+            elif setting == 'source':
+                Device.set_port(self, value)
+            else:
+                if value not in self.Allowed_Resolutions:
+                    raise ValueError('This is not an allowed resolution')
+                self.Resolution = value
+        self.Session_Data[setting] = value
+
+
+@dataclass
+class Recording(Session):
```

Comment:
I know this is WIP, but is there any reason we can't use the BioPlusRecording class I created the placeholder for? Might just be a question of patterns, i.e. decoupling the above classes and adding them to the BioPlusRecording class as an additional class property.

Reply:
This is for integrating an interely different api that was impinged on me last second with entirely different requirements, so I thought it best making this first and when I can approach something that your class can use integrate it in

Reply:
ellaborated reponse above